### ISSUE TITLE
Websocket support

### ISSUE THREAD

**Issue by jpalvarezl** [microsoft/typespec#4124]
### Clear and concise description of the problem

As discussed in the meeting, I am opening an issue where we can start a discussion on the design how to best approach what support for websockets in TSP would look like and what are the downstream emitter implications. 

cc @johanste , @bterlson , @trrwilson

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://typespec.io/docs/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by Kreash** [comment]
It might be suitable to use as an [AsyncAPI](https://www.asyncapi.com/) base

**Comment by TertiumOrganum1** [comment]
> It might be suitable to use as an [AsyncAPI](https://www.asyncapi.com/) base

Agree, most services use async connection and/or ws, it's very strange that there is no asyncapi support in typespec yet

**Comment by jpalvarezl** [comment]
With the release as well of the [OpenAI realtime API](https://openai.com/index/introducing-the-realtime-api/) it would be great to have the tooling ready to handle this formally in TypeSpec.
